gc3apps/bf.uzh.ch/gcsvBulk.py

```diff
@@ -0,0 +1,216 @@
+            outputs = gc3libs.ANY_OUTPUT,
+            stdout = 'gcsvBulk.log',
+            join=True,
+            executables = "./%s" % os.path.basename(input_function),
+            **extra_args)
+
+class GcsvBulkScript(SessionBasedScript):
+    """
+    The script takes as input a mfunction, the number of parallel jobs to run and the
+    dataset to be processed
+
+    The ``gcsvBulk`` command keeps a record of jobs (submitted, executed
+    and pending) in a session file (set name with the ``-s`` option); at
+    each invocation of the command, the status of all recorded jobs is
+    updated, output from finished jobs is collected, and a summary table
+    of all known jobs is printed.
+
+    Options can specify a maximum number of jobs that should be in
+    'SUBMITTED' or 'RUNNING' state; ``gcsvBulk`` will delay submission of
+    newly-created jobs so that this limit is never exceeded.
+    """
+
+    def __init__(self):
+        SessionBasedScript.__init__(
+            self,
+            version = __version__, # module version == script version
+            application = GcsvBulkApplication,
+            # only display stats for the top-level policy objects
+            # (which correspond to the processed files) omit counting
+            # actual applications because their number varies over
+            # time as checkpointing and re-submission takes place.
+            stats_only_for = GcsvBulkApplication,
+            )
+
+    def setup_args(self):
+        self.add_param('mfunction', help=(
+            "Name of the matlab function to run."
+            "A corresponding `.m` file must exist in the current directory."))
+        self.add_param('csvfile', help=(
+            "Name of the csv filte to be processed."))
+        self.add_param('N', default=0, type=int, help=("Value for the number of nodes."))
+
+    def get_function_name_and_file(self, mfunction):
+        if mfunction.endswith('.m'):
+            mfunction_m = mfunction
+            mfunction = mfunction[:-len('.m')]
+        else:
+            mfunction_m = mfunction + '.m'
+        m_file = join(os.getcwd(), mfunction_m)
+        if not exists(m_file):
+            raise ArgumentError(
+                self.actions['mfunction'],
+                ("Cannot read file '{m_file}'"
+                 " providing matlab function '{mfunction}'.")
+                .format(**locals()))
+        return mfunction, m_file
+
+    def get_csv_file(self, csvFile):
+        if not csvFile.endswith('.csv'):
+            csvFile += '.csv'
+        csv_file = join(os.getcwd(), csvFile)
+        if not exists(csv_file):
+            raise ArgumentError(
+                self.actions['csvfile'],
+                ("Cannot read file '{csv_file}.")
+                .format(**locals()))
+        return csv_file
+
+    def get_BlockSize(self, nrBlocks, csvFile):
+        with open(csvFile) as f:
+            for i, l in enumerate(f, 1):
+                pass
+        return ceil(i/nrBlocks)
```

Comment:
Unless you really want to return an integer cast as a floating-point number, there's no need for `ceil()` here -- since both `i` and `nrBlocks` are integers, `i//nrBlocks` will be the result of integer division.